internal/pkg/api/handleUpload.go

```diff
@@ -37,72 +34,7 @@ const (
 	maxUploadTimer = 24 * time.Hour
 )
 
-func (rt Router) handleUploadStart(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
-	start := time.Now()
-	reqID := r.Header.Get(logger.HeaderRequestID)
-	zlog := log.With().
-		Str(ECSHTTPRequestID, reqID).
-		Logger()
-
-	// authentication occurs inside here
-	// to check that key agent ID matches the ID in the body payload yet-to-be unmarshalled
-	if err := rt.ut.handleUploadStart(&zlog, w, r); err != nil {
-		writeUploadError(err, w, zlog, start, "error initiating upload process")
-		return
-	}
-}
-
-func (rt Router) handleUploadChunk(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
-	start := time.Now()
-
-	id := ps.ByName("id")
-	chunkID := ps.ByName("num")
-
-	reqID := r.Header.Get(logger.HeaderRequestID)
-
-	zlog := log.With().
-		Str(LogAgentID, id).
-		Str(ECSHTTPRequestID, reqID).
-		Logger()
-
-	// simpler authentication check,  for high chunk throughput
-	// since chunk checksums must match transit hash
-	// AND optionally the initial hash, both having stricter auth checks
-	if _, err := rt.ut.authAPIKey(r, rt.bulker, rt.ut.cache); err != nil {
-		writeUploadError(err, w, zlog, start, "authentication failure for chunk write")
-		return
-	}
-
-	chunkNum, err := strconv.Atoi(chunkID)
-	if err != nil {
-		writeUploadError(uploader.ErrInvalidChunkNum, w, zlog, start, "error parsing chunk index")
-		return
-	}
-	if err := rt.ut.handleUploadChunk(&zlog, w, r, id, chunkNum); err != nil {
-		writeUploadError(err, w, zlog, start, "error uploading chunk")
-		return
-	}
-}
-
-func (rt Router) handleUploadComplete(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
-	start := time.Now()
-
-	id := ps.ByName("id")
-
-	reqID := r.Header.Get(logger.HeaderRequestID)
-
-	zlog := log.With().
-		Str(LogAgentID, id).
-		Str(ECSHTTPRequestID, reqID).
-		Logger()
-
-	// authentication occurs inside here, to ensure key agent ID
-	// matches the same agent ID the operation started with
-	if err := rt.ut.handleUploadComplete(&zlog, w, r, id); err != nil {
-		writeUploadError(err, w, zlog, start, "error finalizing upload")
-		return
-	}
-}
+// FIXME Should we use the structs in openapi.gen.go instead of the generic ones? Will need to rework the uploader if we do
```

Comment:
@pzl, The openapi spec defines the `upload_id` params as UUIDs and is strict about checking them.
I've altered the endpoint/uploader implementation to support this and use the generated structs where I could.

The only endpoint that does not use a generated struct at the moment is `uploadBegin` as it would require more changes in the uploader implementation.

I've also added metrics tracking across all 3 upload endpoints

Reply:
thanks!

The upload-id doesn't necessarily _have_ to be a UUID. The alternative would be to keep everything as opaque `string` and just change the openapi to be less specific. No issues with going either way

Reply:
I've altered the openapi spec to remove the UUID requirement